### PR TITLE
feat: add referrals page

### PR DESCRIPTION
**Pull request by travis**
Based on Figma mockups. Uses a simple form to send users to the console - VUE_APP_CONSOLE_URL should be set to https://console.storacha.network in production.

A couple known TODOs:

- [x] update TOS link
- [ ] set environment variable in staging/production

Screenshot 2024-11-20 at 2 20 16 PM
Screenshot 2024-11-20 at 2 20 32 PM
Screenshot 2024-11-20 at 2 21 03 PM
Screenshot 2024-11-20 at 2 21 23 PM
Screenshot 2024-11-20 at 2 21 31 PM

### CODE REVIEW

File: app.config.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export default defineAppConfig({
       { text: 'Node Providers', href: '/node-providers' },
       { text: 'Blog', href: '/blog' },
       { text: 'Docs', href: 'https://docs.storacha.network' },
+      { text: 'Referrals', href: 'https://storacha.network/referrals' },
```
Comment (cwaring) [inline app.config.ts:20]:
```suggestion
      { text: 'Referrals', href: '/referrals' },
```
internal app pages should be relative

File: components/ReferralCodeCreator.vue [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<script setup lang="ts">
+export interface ReferralCodeCreatorProps {
+  referralsUrl: string
+}
+
+const referralsUrl =  `${process.env.VUE_APP_CONSOLE_URL}/referrals`
```
Comment (cwaring) [inline components/ReferralCodeCreator.vue:6]: `process.env` will not be available for es modules after build, you should use `import.meta.env` instead where required: https://nuxt.com/docs/api/advanced/import-meta#builder-properties

**Review by cwaring — CHANGES_REQUESTED** [review]
hey @travis, I spotted a couple of errors here and added some review notes.

also for consistency and responsive behaviour I'd suggest using the other frame of the body split component to hold the form element instead of the header, but ultimately a stylistic choice here :)

**Comment by travis** [comment]
Thank you @cwaring !

> other frame of the body split component to hold the form element instead of the header, but ultimately a stylistic choice here :)

not fully following here - are you suggesting I move `ReferralCodeCreator` out of the `SectionHeader` `actions` and instead use a new `Split` component below the `HeroBase` on that page?

**Review by fforbeck — APPROVED** [review]
LGTM!

**Pull request merged**